src/devtools/cache.js

```diff
@@ -1,6 +1,7 @@
 // @flow
 
 import React, { createContext } from 'react';
+import LRU from 'lru-cache';
```

Comment:
This complicates the `cache` definition a bit, but I think it's necessary to avoid leaking memory over time as we inspect a lot of elements.

Alternately...maybe I could use a `WeakMap` and use the `Element` object in the `Store` as the key? Thoughts?

Reply:
Do we promise to never recreate elements? I’m not sure we rely on their identity anywhere. I guess there’s no reason to recreate them..

Reply:
We create them on mount and otherwise mutate them for e.g. changes in `isCollapsed`, `weight`, or `children`

Reply:
Btw was always curious — is there any reason we recreate children array instead of mutating them on changes?

Reply:
Yes, but maybe it's a premature optimization 😅 

https://github.com/bvaughn/react-devtools-experimental/blob/8b33dd673d965d23d2516bd30c9c3d05386ce881/src/devtools/store.js#L71-L73

Reply:
Regarding mutable elements- maybe it's a premature optimization 😅but it seemed worth  doing

https://github.com/bvaughn/react-devtools-experimental/blob/8b33dd673d965d23d2516bd30c9c3d05386ce881/src/devtools/store.js#L71-L73

> is there any reason we recreate children array instead of mutating them on changes?

Honestly? I think laziness.

Reply:
I went ahead and made this change. Feels nicer.